src/style-property-map.js

```diff
@@ -36,22 +36,58 @@
     }
 
     if (!cssPropertyDictionary.isValidInput(property, value)) {
-      if (value instanceof KeywordValue) {
-        throw new TypeError(property +
-          ' does not take the keyword ' + value.cssString);
-      }
-      throw new TypeError(property +
-        ' does not take values of type ' + value.constructor.name);
+      cssPropertyDictionary.throwInvalidInputError(property, value);
     }
     this._styleObject[property] = value.cssString;
   };
 
-  StylePropertyMap.prototype.append = function(property, value) {
-    throw new TypeError('Function not implemented yet');
+  StylePropertyMap.prototype.append = function(property, values) {
+    var cssPropertyDictionary = propertyDictionary();
+    if (!cssPropertyDictionary.isSupportedProperty(property)) {
+      throw new TypeError(property + ' is not a supported CSS property');
+    }
+
+    if (!cssPropertyDictionary.isListValuedProperty(property)) {
+      throw new TypeError(property + ' does not support sequences of styleValues');
+    }
+
+    if (values == null) {
+      throw new TypeError('null cannot be appended to CSS properties');
+    }
+
+    if (!(values instanceof Array)) {
+      values = [values];
+    }
+
+    var cssAppendString = this._styleObject[property];
+    var valueSeparator = cssPropertyDictionary.getListValueSeparator(property);
+    for (var i = 0; i < values.length; i++) {
+      if (!cssPropertyDictionary.isValidInput(property, values[i])) {
+        cssPropertyDictionary.throwInvalidInputError(property, values[i]);
+      }
+      if (cssAppendString == '') {
+        cssAppendString += values[i].cssString;
+        continue;
+      }
+      cssAppendString += valueSeparator + values[i].cssString;
+    }
+    return this._styleObject[property] = cssAppendString;
   };
 
   StylePropertyMap.prototype.delete = function(property) {
-    throw new TypeError('Function not implemented yet');
+    var cssPropertyDictionary = propertyDictionary();
+    if (!cssPropertyDictionary.isSupportedProperty(property)) {
+      throw new TypeError(property + ' is not a supported CSS property');
+    }
+    this._styleObject[property] = '';
+  };
+
+  StylePropertyMap.prototype.has = function(property) {
+    var cssPropertyDictionary = propertyDictionary();
+    if (!cssPropertyDictionary.isSupportedProperty(property)) {
+      throw new TypeError(property + ' is not a supported CSS property');
+    }
+    return !(this._styleObject[property] == '');
```

Comment:
You can just write
```javascript
return !!this._styleObject[property]
```
The "", null, undefined and 0 all evaluate to false, most other things evaluate to true, and using `!!` coerces the value to be a boolean.